From 42d60e18a1ccf20ccc25787da643f197b55a0a4a Mon Sep 17 00:00:00 2001
From: Grygorii Strashko <grygorii.strashko@ti.com>
Date: Tue, 22 Dec 2020 13:04:22 +0200
Subject: [PATCH 02/35] net: ethernet: ti: icss_iep: disable extts and pps if
 no slow compensation or non 64bit mode

Disable extts and pps if no slow compensation support or non 64bit mode.

Signed-off-by: Grygorii Strashko <grygorii.strashko@ti.com>
---
 drivers/net/ethernet/ti/icss_iep.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/ethernet/ti/icss_iep.c b/drivers/net/ethernet/ti/icss_iep.c
index db8cf28acd13..0b240e9c6908 100644
--- a/drivers/net/ethernet/ti/icss_iep.c
+++ b/drivers/net/ethernet/ti/icss_iep.c
@@ -734,6 +734,11 @@ struct icss_iep *icss_iep_get(struct device_node *np)
 
 	iep->ptp_info = icss_iep_ptp_info;
 
+
+	if (!(iep->plat_data->flags & ICSS_IEP_64BIT_COUNTER_SUPPORT) ||
+	    !(iep->plat_data->flags & ICSS_IEP_SLOW_COMPEN_REG_SUPPORT))
+		goto exit;
+
 	if (iep->cap_cmp_irq || (iep->ops && iep->ops->perout_enable)) {
 		iep->ptp_info.n_per_out = 1;
 		iep->ptp_info.pps = 1;
@@ -742,6 +747,7 @@ struct icss_iep *icss_iep_get(struct device_node *np)
 	if (iep->cap_cmp_irq || (iep->ops && iep->ops->extts_enable))
 		iep->ptp_info.n_ext_ts = 2;
 
+exit:
 	return iep;
 
 put_iep_device:
-- 
2.17.1

